File: Fortran/1_helloMPI/solution/hello.f90
program hello
    ! TODO: Make the MPI functions and constants available
    use mpi
    integer :: ierr, numProcs, rank 
    ! TODO: Initialise the MPI library
    call MPI_INIT(ierr)
    ! TODO: Obtain the process ID and the number of processes
    call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierr)
    call MPI_COMM_SIZE(MPI_COMM_WORLD, numProcs, ierr)
    ! TODO: Display the process ID and the number of processes
    write(*, fmt="(a, i2, a, i2, a)") "Hello world from rank ", rank, " of ", numProcs, " in communicator MPI_COMM_WORLD."

    ! TODO: Finalize the MPI library
    call MPI_FINALIZE(ierr)
end program hello

File: C/1_helloMPI/hello.c
#include <stdio.h>
#include <stdlib.h>

// TODO: Include MPI header to declare MPI prototypes and con
#include <mpi.h>

int main (int argc, char* argv[])
{
    // TODO: Initialise the MPI library
    MPI_Init(&argc, &argv);

    // TODO: Obtain the process ID and the number of processes
    int worldrank, worldsize;
    MPI_Comm_rank(MPI_COMM_WORLD, &worldrank);
    MPI_Comm_size(MPI_COMM_WORLD, &worldsize);

    // TODO: Display the process ID and the number of processes
    printf("Hello world from rank %d of %d in communicator.\n", worldrank, worldsize);

    // TODO: Finalize the MPI library
    MPI_Finalize();

    return EXIT_SUCCESS;
}
